applicationLibrary/molecularDynamics/molecularDynamicsLibrary/AxilrodTellerMutoFunctorHWY.h

```diff
@@ -0,0 +1,1836 @@
+      const size_t numKernelCallsNoN3Acc =
+          std::accumulate(_aosThreadDataFLOPs.begin(), _aosThreadDataFLOPs.end(), 0ul,
+                          [](size_t sum, const auto &data) { return sum + data.numKernelCallsNoN3; });
+
+      return (static_cast<double>(numKernelCallsNoN3Acc) + static_cast<double>(numKernelCallsN3Acc)) /
+             (static_cast<double>(numTripletsCount));
+    } else {
+      // This is needed because this function still gets called with FLOP logging disabled, just nothing is done with it
+      return std::numeric_limits<double>::quiet_NaN();
+    }
+  }
+
+ private:
+  template <bool LowerTriangle, bool alignedSoAView>
+  struct DistanceMatrix;
+
+  template <bool alignedSoAView>
+  void SoAFunctorSingleImpl(autopas::SoAView<SoAArraysType> soa) {
+    const auto threadnum = autopas::autopas_get_thread_num();
+
+    const auto *const __restrict xPtr = soa.template begin<Particle_T::AttributeNames::posX>();
+    const auto *const __restrict yPtr = soa.template begin<Particle_T::AttributeNames::posY>();
+    const auto *const __restrict zPtr = soa.template begin<Particle_T::AttributeNames::posZ>();
+    const auto *const __restrict ownedStatePtr = soa.template begin<Particle_T::AttributeNames::ownershipState>();
+
+    SoAFloatPrecision *const __restrict fxPtr = soa.template begin<Particle_T::AttributeNames::forceX>();
+    SoAFloatPrecision *const __restrict fyPtr = soa.template begin<Particle_T::AttributeNames::forceY>();
+    SoAFloatPrecision *const __restrict fzPtr = soa.template begin<Particle_T::AttributeNames::forceZ>();
+
+    [[maybe_unused]] auto *const __restrict typePtr = soa.template begin<Particle_T::AttributeNames::typeId>();
+
+    // the local redeclaration of the following values helps the SoAFloatPrecision-generation of various compilers.
+    const SoAFloatPrecision cutoffSquared = _cutoffSquared;
+
+    VectorDouble potentialEnergySum =
+        _zeroDoubleVec;  // Note: This is not the potential energy but some fixed multiple of it.
+    VectorDouble virialSumX = _zeroDoubleVec;
+    VectorDouble virialSumY = _zeroDoubleVec;
+    VectorDouble virialSumZ = _zeroDoubleVec;
+
+    size_t numTripletsCountingSum = 0;
+    size_t numDistanceCalculationSum = 0;
+    size_t numKernelCallsN3Sum = 0;
+    size_t numKernelCallsNoN3Sum = 0;
+    size_t numGlobalCalcsN3Sum = 0;
+    size_t numGlobalCalcsNoN3Sum = 0;
+
+    const SoAFloatPrecision const_nu = _nu;
+    const size_t soaSize = soa.size();
+
+    if constexpr (countFLOPs) {
+      numTripletsCountingSum = soaSize * (soaSize - 1) * (soaSize - 2) / 6;
+    }
+    // Precompute distances between particles in the soa. We use a row-major packed lower‐triangle data structure to
+    // save memory. If a distance i->j is required as j->i it can be negated.
+    DistanceMatrix<true, alignedSoAView> intraSoADists(soaSize, soaSize, precomputeBuffer1[threadnum]);
```

Comment:
```suggestion
    DistanceMatrix<true, alignedSoAView> intraSoAPairDists(soaSize, soaSize, precomputeBuffer1[threadnum]);
```
or something else that makes it clear that this is pairwise distances.